File: framework/extracted/Loenn/entities/shadowFigure.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local shadowFigure = {}

shadowFigure.name = "MaggyHelper/ShadowFigure"
shadowFigure.depth = -50
shadowFigure.fieldInformation = {
    detectionRange = {
        minimumValue = 0
    },
    followDistance = {
        minimumValue = 0
    }
}
shadowFigure.placements = {
    {
        name = "passive",
        data = {
            detectionRange = 150,
            followDistance = 80,
            isHostile = false
        }
    },
    {
        name = "hostile",
        data = {
            detectionRange = 150,
            followDistance = 80,
            isHostile = true
        }
    }
}

function shadowFigure.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("characters/shadow_figure/hidden", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    sprite.color = {0.0, 0.0, 0.0, 0.7}
    return sprite
end

function shadowFigure.selection(room, entity)
    return utils.rectangle(entity.x - 12, entity.y - 48, 24, 48)
end

return shadowFigure

File: framework/extracted/Ahorn/entities/ShadowFigure.jl
module ShadowFigure

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/ShadowFigure" ShadowFigure(x::Integer, y::Integer, detectionRange::Integer=150, followDistance::Integer=80, isHostile::Bool=false)

const placements = Ahorn.PlacementDict(
    "passive" => Ahorn.EntityPlacement(ShadowFigure),
    "hostile" => Ahorn.EntityPlacement(ShadowFigure)
)

function Ahorn.selection(entity::ShadowFigure)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::ShadowFigure, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end
